product/views.py

```diff
@@ -73,5 +72,95 @@ def get(self, request):
+            shoe_detail.append({
+                'sub_image'  : sub_image,
+                'color_list' : color_list,
+                'size_list'  : size_list
+            })
+            return JsonResponse({'product' : shoe_detail}, status=200)
+        except IndexError:
+            return JsonResponse({'message' : 'NON_EXISTING_PRODUCT'}, status=400)
+        except ValueError:
+            return JsonResponse({'message' : 'VALUE_ERROR'}, status=400)
+
+class FilterView(View):
+    def get(self, request):
+        page = int(request.GET.get('page', 0))
+        limit = int(request.GET.get('limit', 20))
+        colorfilter  = request.GET.getlist('color', list(ColorFilter.objects.all().values_list('name', flat=True)))
+        typefilter   = request.GET.getlist('type', list(TypeFilter.objects.all().values_list('name', flat=True)))
+        genderfilter = request.GET.getlist('gender', list(GenderSegmentation.objects.all().values_list('name', flat=True)))
+        sizefilter   = request.GET.getlist('size', list(Size.objects.all().values_list('name', flat=True)))
+        
+        shoes = ShoeColor.objects.filter(**{
+            'color__color_category__name__in'     : colorfilter,
+            'shoe__type_filter__name__in'         : typefilter,
+            'shoe__gender_segmentation__name__in' : genderfilter,
+            'shoecolorsize__size__name__in'       : sizefilter,
+            'subimage__is_hovered'                : True
+        }).distinct()
         
-        return JsonResponse({'products' : shoe_list}, status=200)
+        shoe_values = list(shoes.annotate(
+            name       = F('shoe__detail__name'),
+            price      = F('shoe__price'),
+            main_image = F('image__image'),
+            sub_image  = F('subimage__image')
+        ).values('id','shoe__id', 'name', 'price', 'main_image', 'sub_image')[page*limit:((page+1)*limit)-1])
+
+        filters = {
+            'gender_filters'    : list(shoes.values_list('shoe__gender_segmentation__name',flat=True)),
+            'color_filters'     : [color.name for color in ColorFilter.objects.all()],
+            'type_filters'      : list(shoes.values_list('shoe__type_filter__name',flat=True)),
+            'size_filters'      : list(shoes.values_list('shoecolorsize__size__name',flat=True))
+        }
+
+        shoe_list = [{'product_detail' : shoe} for shoe in shoe_values]
+        
+        for shoe in shoe_list:
+            shoe['color_list'] = list(Color.objects.filter(**{
+                'shoecolor__shoe__id'             : shoe['product_detail']['shoe__id'],
+                'shoecolor__subimage__is_hovered' : True
+            }).annotate(
+                shoe_id      = F('shoecolor__id'),
+                color_filter = F('color_category__name'),
+                main_image   = F('shoecolor__image__image'),
+                sub_image    = F('shoecolor__subimage__image')
+            ).values('shoe_id', 'color_filter', 'main_image', 'sub_image'))
```

Comment:
코드 라인도 훨씬 줄뿐더러 위에서 prefetch 또는 select_related 로 쿼리하시면 쿼리수 도 줄어 듭니다.
이부분 그때 알려드린 쿼리수와 실행시간 성능측정 데코레이터사용하셔서 확인해보시길 권장 드립니다.
```suggestion
                shoe_list = [{
            'product_detail' : {
                'id' : shoe.id,
                'name' : shoe.shoe.detail.name,
                'price' : shoe.shoe.price
            }
        } for shoe in shoes]

Reply:
annotate는 사용하지 않았는데, list comprehension으로 for loop 두개를 합치고 싶었으나, 이미 생성된 객체의 value를 사용해야 하는부분이 있어서 합쳐지질 않아요...!